### ISSUE TITLE
Keyboard trap in playground version of carousel

### ISSUE THREAD

**Issue by cstrobbe** [ec-europa/europa-component-library#2967]
### Actual behaviour

After tabbing to the last button in the carousel (i.e. the last round button of the row at the bottom), the next press of the TAB key does not focus to the next item on the page but back to the first button (i.e. the "Previous slides" button).
This affects the playground versions of the carousel ([EC](https://ec.europa.eu/component-library/playground/ec/?path=/story/components-carousel--default) and [EU](https://ec.europa.eu/component-library/playground/eu/?path=/story/components-carousel--default)) but not the versions in the [showcase](https://ec.europa.eu/component-library/ec/components/carousel/usage/).

This may be due to the `keydown` event listener on the element `<div class="ecl-carousel__navigation">`, which wraps around the navigation bottoms at the button, and which contains code such as the following to move the focus to the "Previous slides" button:

```JavaScript
  switch (e.key) {
    case "Tab":
    case "ArrowRight":
      e.preventDefault(), (t.nextSibling ? (this.shiftSlide("next", !0), t.nextSibling) : this.btnPrev).focus();
      break;
    case "ArrowLeft":
      (t.previousSibling ? (this.shiftSlide("prev", !0), t.previousSibling) : this.btnPlay).focus()
  }
```

When tabbing backwards (with Shift Tab, starting from e.g. the tab "CSS Resources), there is no keyboard trap but the keyboard focus does not move through all the buttons.

### Expected behaviour

1. When tabbing forwards, after reaching the "Go to slide 3" button, pressing the TAB key again should simply move to the next focusable element on the page.
2. When tabbing backwards, the keyboard focus should move through all the buttons / focusable elements in the carousel.

### Steps to reproduce the issue

1. Open the carousel example in the playground and press TAB to navigate through it. After reaching the button "Next slides", the keyboard focus moves to the button "Previous slides".
2. In the same example, move the focus backwards using Shift TAB  (for efficiency, right-click on the tab "CSS Resources", then dismiss the context menu) and notice that not all buttons in the carousel receive keyboard focus.

**Comment by planctus** [comment]
confirmed, although the behavior, when navigating toward the "bullets" is by requirements, i mean the fact that from the play you go to the selected slide, whatever that is, might feel unexpected, but that is not a bug,  but it's true that there is this trap in lage viewports, we will fix this in the next release.

**Comment by cstrobbe** [comment]
> the fact that from the play you go to the selected slide, whatever that is, might feel unexpected, but that is not a bug

It is absolutely a bug. Imagine you have a carousel like this at the top of the page and a lot of content below it. In that situation, a keyboard can never reach the information below the carousel by tabbing forwards. Most accessibility auditors will mark this as a failure of [Success Criterion 2.1.2 No Keyboard Trap](https://www.w3.org/TR/WCAG21/#no-keyboard-trap), which is one of the [non-interference criteria](https://www.w3.org/TR/WCAG21/#cc5). This should be treated as a serious issue.

**Comment by planctus** [comment]
it seems i wasn't clear, the "trap" is surely a bug and will be fixed in the next release, we have already created a ticket about it. What i explained in the previous comment is that tabbing from the "play" button moves the focus to the bullet corresponding to the active slide, not the first one, which might be unexpected, but as i wrote, that is how it was requested.
Thank you

**Comment by planctus** [comment]
You can already see the problem fixed here: https://ecl-preview-2968--europa-component-library.netlify.app/playground/ec/?path=/story/components-carousel--default

**Comment by planctus** [comment]
@cstrobbe, since this was merged and it's going to be released in the next ECL release, may i ask you to take a look at the component and confirm that the issue is fixed? In case it is i guess we can also close this. Thanks!
https://v3-dev--europa-component-library.netlify.app/playground/ec/?path=/story/components-carousel--default

**Comment by cstrobbe** [comment]
In the version on netlify, there is no longer a keyboard trap when tabbing forwards, but there is still a sort of trap when tabbing backwards: when the keyboard focus is on the button for the first slide ("Go to slide 1"), pressing Shift Tab moves between the Play button and the "Go to slide 1" button, with no way to get past these button using Shift Tab. (This applies both with and without a screen reader.) This may be related to the keydown listener on the Play button.

**Comment by planctus** [comment]
Alright, thanks for checking the current status, indeed we handled shift+tab in the "bullets" but not in the play button, this has been added here: https://ecl-preview-2979--europa-component-library.netlify.app/playground/ec/?path=/story/components-carousel--default
